opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala

```diff
@@ -31,7 +31,22 @@ object Tracing {
   private def createChildOf(parent: Span, spanName: String, spanKind: SpanKind): URManaged[Tracing, Span] =
     ZManaged.accessManaged[Tracing](_.get.createChildOf(parent, spanName, spanKind))
 
-  private def getCurrentSpan: URIO[Tracing, Span] = currentSpan.flatMap(_.get)
+  /**
+   * Get the current span stored in the zio-telemetry ref
+   * @return
+   */
+  def getCurrentSpan: URIO[Tracing, Span] = currentSpan.flatMap(_.get)
+
+  /**
+   * Get the current span stored in the zio-telemetry ref, allowing
+   * integration with externally created spans from native
+   * opentelemetry-api instrumentations
+   *
+   * @param span The span to set as current
+   * @return
+   */
+  def setCurrentSpan(span: Span): URIO[Tracing, Unit] =
```

Comment:
Would it make sense to stop the previous span or to restore it later?

Reply:
I'm not sure. The use case I've had for this is integrating with spans created via the play and akka http instrumentation, so we'd only be replacing the default span. There might be a better way to "re-initialize" the span via the zio runtime, too, but I didn't have much luck with it.